core/src/main/java/org/apache/calcite/adapter/enumerable/EnumerableHashJoin.java

```diff
@@ -229,7 +229,10 @@ private Result implementHashJoin(EnumerableRelImplementor implementor, Prefer pr
                         Expressions.constant(joinType.generatesNullsOnLeft()))
                     .append(
                         Expressions.constant(
-                            joinType.generatesNullsOnRight())))).toBlock());
+                            joinType.generatesNullsOnRight()))
+                    .append(
+                        Expressions.constant(
```

Comment:
Minor detail. We could simplify this expression with Expressions.list and four append:
```
Expressions.list(
    A, B, C, D)
.append(E)
.append(F)
.append(G)
.append(H)
```
by just moving the append parameters directly inside the Expressions.list (and get rid of all the append calls):
```
Expressions.list(
    A, B, C, D, E, F, G, H)
```

Reply:
I agree, thanks.